explorer/interpreter/impl_scope.cpp

```diff
@@ -225,13 +226,32 @@ static auto CombineResults(Nonnull<const InterfaceType*> iface_type,
   if (!impl_a) {
     return b;
   }
+
   // If they refer to the same `impl` declaration, it doesn't matter which one
   // we pick, so we pick `a`.
   // TODO: Compare the identities of the `impl`s, not the declarations.
   if (&impl_a->declaration() == &impl_b->declaration()) {
     return a;
   }
+
   // TODO: Order the `impl`s based on type structure.
+
+  // If the declarations appear in the same `match_first` block, whichever
+  // appears first wins.
+  // TODO: Once we support an impl being declared more than once, we will need
+  // to check this more carefully.
+  if (impl_a->declaration().match_first() &&
```

Comment:
I'll note, a consequence of this particular syntax choice is that an external impl cannot be disambiguated in more than one priority ordering. i.e., if there is any ambiguity, it must be provide priorities for _all_ ambiguities. I'm not sure that the "declared more than once" comment addresses that issue, since couldn't an external impl be prioritized versus an impl in a different, imported package? (note, I'm particularly uncertain on the last since I know there are planned restrictions, but I'm not sure)

Reply:
`impl` declarations in a `match_first` are "real" declarations of the `impl`, so they can only appear within the library that "owns" that `impl`. You can't use `match_first` to prioritize across libraries, but it wouldn't make sense to do so anyway because `match_first` only disambiguates between` impl`s with the same type structure, and there's at most one library in which an `impl` with a given type structure can appear. (Nearly all of this is not yet implemented; `match_first` syntax is just an early step into this `impl` prioritization system.)